<?xml version="1.0" encoding="utf-8"?>
<launch>

    <arg name="robot_name" />

    <!-- Handle mapping -->
	<include file="$(find rhoa_sim)/launch/rhoa_gmapping.launch">
		<arg name="robot_name" value="$(arg robot_name)" />
	</include>

    <!-- Launch aruco code detection -->
    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher" output="log" ns="$(arg robot_name)">
        <remap from="/camera_info"       to="camera/front/camera_info" />
        <remap from="/image"             to="camera/front/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="0.15"/>
        <param name="reference_frame"    value=""/>
        <param name="camera_frame"       value="$(arg robot_name)_tf/front_camera"/>
    </node>

    <node pkg="rhoa_sim" type="knowledge_handler.py" name="knowledge_handler" output="log" ns="$(arg robot_name)">
    	<param name="robot_name"  value="$(arg robot_name)"/>
    	<param name="rate"        value="5"/>
    </node>
</launch>

